File: C#/SecantMethod/secantmethod.cs
using System;

namespace SecantMethod
{
	class MainClass
	{
		public static void Main (string[] args)
		{
			Console.WriteLine ("x0:");
			double x0 = Convert.ToDouble(Console.ReadLine());
			Console.WriteLine ("x1:");
			double x1 = Convert.ToDouble(Console.ReadLine());
			double x2;
			double eps = 1e-6;
			int i = 0;
			double Dx;

			Console.WriteLine ("i, x0, x1, f (x0), f(x1), x2, Dx");
			do {
				x2 = ((x0 * f (x1)) - (x1 * f (x0))) / (f (x1) - f (x0));
				Dx = Math.Abs (x2 - x1);
				Console.WriteLine ("{0},{1},{2},{3},{4},{5}", i + 1, x0, x1, f (x0), f (x1), x2, Dx);
				x0 = x1;
				x1 = x2;
				i++;
			} while (Dx > eps);
			Console.ReadLine ();
		}

		private static double f(double x)
		{
			return x * Math.Exp (-x) + Math.Pow (x, 3.0) + 1.0;
		}
	}
}

File: Ruby/secantmethod.rb
n, eps = 0, 1e-6
x0 = [(print 'x0: '), gets.rstrip][1].to_f
x1 = [(print 'x1: '), gets.rstrip][1].to_f

def f x
  x*Math.exp(-x)+(x**3.0)+1
end

puts "n   x0     x1    f(x0)    f(x1)   x2    |Dx|"

while true do
  x2 = ((x0 * f(x1)) - (x1 * f(x0))) / (f(x1) - f(x0))
  dx = (x2 - x1).abs
  puts "#{n+1} #{'%.4f' % x0} #{'%.4f' % x1} #{'%.4f' % f(x0)} #{'%.4f' % f(x1)} #{'%.4f' % x2} #{'%.4f' % dx}"
  x0, x1 = x1, x2
  n += 1
  if dx < eps
    break end
end
